core/src/main/java/org/fao/geonet/kernel/security/keycloak/keycloakPreAuthActionsLoginFilter.java

```diff
@@ -58,12 +59,24 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         HttpServletResponse servletResponse = (HttpServletResponse) response;
 
         if (servletRequest.getPathInfo() != null &&
-                !(servletRequest.getContextPath() + KeycloakUtil.getSigninPath()).equals(servletRequest.getRequestURI())  &&
-                !(servletRequest.getPathInfo()).equals("/k_logout")  &&
-                !isAuthenticated() ) {
+            !KeycloakAuthenticationProcessingFilter.DEFAULT_REQUEST_MATCHER.matches(servletRequest) &&
```

Comment:
I would add a comment explaining a bit the checks done, for example about the `AdapterConstants` values checks and in case that makes sense wrap them in a function.